[terra-form-input] Adding doc warning about chromium VO bug (#3840)
cerner/terra-core: packages/terra-core-docs/src/terra-dev-site/doc/form-input/About.1.doc.mdx

**Note:** To follow [UX Best Practices](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Placeholder_Research), this component does not allow `placeholder`.

<Notice variant="caution" ariaLevel="3">

> **Accessibility Bug:** Currently a bug exists in Chromium based browsers with VoiceOver where `<input type="number" />` is read incorrectly when incrementing or decrementing the value with keyboard shortcuts. Terra Form Input utilizes `<input type="number" />` when given "number" as type, and thus will also have this screenreader issue. An alternative is to use something like `<input type="text" inputmode="numeric" pattern="\d*" />`, however this will have no built in validity checks or keyboard commands.

Reviewer: ```suggestion
**Accessibility Bug:** Due to a bug in Chromium-based browsers, VoiceOver would read `<input type="number" />` incorrectly when incrementing or decrementing the value with keyboard shortcuts. Terra Form Input utilizes `<input type="number" />` when the `type` prop is set to "number" and will also have this screenreader issue. An alternative is to use something like `<input type="text" inputmode="numeric" pattern="\d*" />`. However, this will have no built in validity checks or keyboard commands.
```

Reviewer: Did you mean to use `input` or `TerraFormInput` here?
```js
<input type="text" inputmode="numeric" pattern="\d*" />
```

Reviewer 2: ```suggestion
**Accessibility Bug:** Currently a bug exists in Chromium based browsers (Chrome, Microsoft Edge, etc.) with VoiceOver where `<input type="number" />` is read incorrectly when incrementing or decrementing the value with keyboard shortcuts. Terra Form Input utilizes `<input type="number" />` when given "number" as type, and thus will also have this screenreader issue. An alternative is to use something like `<input type="text" inputmode="numeric" pattern="\d*" />`, however this will have no built-in validity checks or keyboard commands. Consumers are responsible for adding this functionality.
```

Author: > Did you mean to use `input` or `TerraFormInput` here?
> 
> ```js
> <input type="text" inputmode="numeric" pattern="\d*" />
> ```

I did mean to use `input`. While i think we would pass those attributes down, I don't know if we want them to use our terra-form-input in that way without a lot of testing.

Change to the paragraph here: [17557c3](https://github.com/cerner/terra-core/pull/3840/commits/17557c30e2583a402cc0755813e17bbd8ba26da6)

> **Accessibility Bug:** Due to a bug in Chromium-based browsers (Chrome, Microsoft Edge, etc.), VoiceOver would read `<input type="number" />` incorrectly when incrementing or decrementing the value with keyboard shortcuts. Terra Form Input utilizes `<input type="number" />` when the `type` prop is set to "number" and will also have this screenreader issue. An alternative is to use something like `<input type="text" inputmode="numeric" pattern="\d*" />`. However, this will have no built in validity checks or keyboard commands. Consumers are responsible for adding this functionality.

Reviewer: Based on [this clarifying comment](https://github.com/cerner/terra-core/pull/3840#discussion_r1258931306 ), I think we should specify that we're using the vanilla html input since w're switching context from the previous sentence. Suggested changes:

````suggestion
**Accessibility Bug:** Due to a bug in Chromium-based browsers (Chrome, Microsoft Edge, etc.), VoiceOver would read `<input type="number" />` incorrectly when incrementing or decrementing the value with keyboard shortcuts. Terra Form Input utilizes `<input type="number" />` when the `type` prop is set to "number" and will also have this screenreader issue. A recommended solution would be to use the vanilla html `input` instead of Terra Form Input. For example: 

 ```html
<input type="text" inputmode="numeric" pattern="\d*" />
```

However, this will not have the built-in validity checks or keyboard commands that Terra Form Input provides. Consumers are responsible for adding this functionality.
````

CC: @eawww for accuracy of the statements.

Reviewer 2: nit/question: Is "vanilla" as a synonym for "ordinary" fairly universal among English speakers from assorted cultural backgrounds? I assumed it was specific to America's specific variety of ice cream obsession.

Author: We could use "basic html"  or "html standard" instead of "vanilla html".

Reviewer: Yeah `basic html` is fine. I said vanilla because vanilla jsavascript is a relatively common term but basic sounds more clearer.

Author: Changed here https://github.com/cerner/terra-core/pull/3840/commits/fbc8fe60c17feb83830edb4adf0f8581b9b84e0f

**Note:** To follow [UX Best Practices](https://www.w3.org/WAI/GL/low-vision-a11y-tf/wiki/Placeholder_Research), this component does not allow `placeholder`.

<Notice variant="caution" ariaLevel="3">

> **Accessibility Bug:** Due to a bug in Chromium-based browsers (Chrome, Microsoft Edge, etc.), VoiceOver would read `<input type="number" />` incorrectly when incrementing or decrementing the value with keyboard shortcuts. Terra Form Input utilizes `<input type="number" />` when the `type` prop is set to "number" and will also have this screenreader issue. A recommended solution would be to instead use Terra Form Input with the `type` prop set to text and the added props of `inputmode` with a value of "numeric" and `pattern` with a regex expressiom that matchs numeric values. For example:

Reviewer 2: ```suggestion
**Accessibility Bug:** Due to a bug in Chromium-based browsers (Chrome, Microsoft Edge, etc.), VoiceOver will read `<input type="number" />` incorrectly when incrementing or decrementing the value with keyboard shortcuts. Terra Form Input utilizes `<input type="number" />` when the `type` prop is set to "number" and will also have this screenreader issue. A recommended solution would be to instead use Terra Form Input with the `type` prop set to "text" and the added props of `inputmode` with a value of "numeric" and `pattern` with a regex expression that matches numeric values. For example:
```
